baybe/utils/optional.py

```diff
@@ -0,0 +1,56 @@
+"""Utilities for handling optional dependencies."""
+
+import importlib
+import importlib.util
+import warnings
+from typing import Any, Literal, Optional
+
+
+def import_optional_module(
+    name: str,
+    attribute: Optional[str] = None,
+    error: Literal["raise", "warn", "ignore"] = "raise",
+) -> Optional[Any]:
+    """Import an optional module.
+
+    Args:
+        name: The name of the module.
+        attribute: The name of an attribute to import from the module.
+        error: How to handle errors.
+            One of:
+                - "raise": Raise an error if the module cannot be imported.
+                - "warn": Raise a warning if the module cannot be imported.
+                - "ignore": Ignore the missing module and return `None`.
+
+    Returns:
+        Optional[Any]: The imported module or attribute from the module, or `None`
```

Comment:
The type hint here should not be necessary.

Reply:
Also, you might need double backtick around `None` for it to be rendered properly.